pkg/cloud/aws/actuators/machine/actuator.go

```
@@ -77,33 +124,31 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
 		return errors.Errorf("failed to retrieve controlplane url during machine creation: %+v", err)
 	}
 
-	var bootstrapToken string
-	if machine.ObjectMeta.Labels["set"] == "node" {
-		kubeConfig, err := a.GetKubeConfig(cluster, nil)
-		if err != nil {
-			return errors.Errorf("failed to retrieve kubeconfig during machine creation: %+v", err)
-		}
-
-		clientConfig, err := clientcmd.BuildConfigFromKubeconfigGetter(controlPlaneURL, func() (*clientcmdapi.Config, error) {
-			return clientcmd.Load([]byte(kubeConfig))
-		})
-
-		if err != nil {
-			return errors.Errorf("failed to retrieve kubeconfig during machine creation: %+v", err)
-		}
+	clusterMachines, err := a.getClusterMachines(scope, cluster)
+	if err != nil {
+		return errors.Errorf("failed to retrieve machines for cluster %q: %v", cluster.Name, err)
+	}
+	controlPlaneMachines := a.getControlPlaneMachines(clusterMachines)
+	isNodeJoin, err := a.isNodeJoin(controlPlaneMachines, machine)
+	if err != nil {
+		return errors.Errorf("Failed to determine whther machine %q should join cluster %q: %v", machine.Name, cluster.Name, err)
+	}
 
-		coreClient, err := corev1.NewForConfig(clientConfig)
+	bootstrapToken := ""
+	if isNodeJoin {
+		bootstrapToken, err = a.getNodeJoinToken(cluster, controlPlaneURL)
 		if err != nil {
-			return errors.Errorf("failed to initialize new corev1 client: %+v", err)
+			klog.Errorf("failed to retrieve token to create machine %q: %v", machine.Name, err)
```

I don't think we need klog here, we can instead return the error